application/app/services/nav/menu_item.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+module Nav
+  class MenuItem
+    attr_reader :id, :label, :url, :position, :hidden, :type, :new_tab, :icon, :partial
+
+    def initialize(id:, label: nil, url: nil, position: nil, hidden: nil, type: nil, new_tab: nil, icon: nil, partial: nil)
+      @id = id
+      @label = label
+      @url = url
+      @position = position
+      @hidden = hidden.nil? ? false : hidden
+      @type = type || infer_type
+      @new_tab = new_tab.nil? ? false : new_tab
+      @icon = icon
+      @partial = partial
```

Comment:
we should validate that the partial path is part of the Rails project and not a malicious path accessing unauthorized files, because this value is injected directly into the view to be rendered

Reply:
same as above